### PR TITLE
Add `fmaf128`

### PR DESCRIPTION
**Pull request by tgross35**
Resolve all remaining `f64`-specific items in the generic version of `fma`, then expose `fmaf128`.

### CODE REVIEW

**Comment by tgross35** [comment]
Not quite

```
---- mp_extensive_fmaf128 ----

    input:    (0xbfe61966cc01966cc01966cc01966f06, 0x8019669933fe69933fe69933fe6997c9, 0x8000000000000000000000000000048a)
    as hex:   (-0x1.1966cc01966cc01966cc01966f06p-25, -0x1.669933fe69933fe69933fe6997c9p-16358, -0x0.000000000000000000000000048ap-16382)
    as bits:  (0xbfe61966cc01966cc01966cc01966f06, 0x8019669933fe69933fe69933fe6997c9, 0x8000000000000000000000000000048a)
    expected: 0x0000c5171470a3ff5e0f68d751491b18 0x0.c5171470a3ff5e0f68d751491b18p-16382 0x0000c5171470a3ff5e0f68d751491b18
    actual:   0x800075d1d71eb80143e12e515d6dc9d0 -0x0.75d1d71eb80143e12e515d6dc9d0p-16382 0x800075d1d71eb80143e12e515d6dc9d0

Caused by:
    mismatched signs 0xbfff0000000000000000000000000000 0x3fff0000000000000000000000000000


failures:
    mp_extensive_fmaf128
```

File: src/math/generic/fma.rs [hunk 1]
```diff
@@ -99,110 +133,159 @@ where
...
 
 /// Representation of `F` that has handled subnormals.
+#[derive(Clone, Copy, Debug)]
 struct Norm<F: Float> {
-    /// Normalized significand with one guard bit.
+    /// Normalized significand with one guard bit, unsigned.
     m: F::Int,
-    /// Unbiased exponent, normalized.
+    /// Exponent of the mantissa such that `m * 2^e = x`. Accounts for the shift in the mantissa
+    /// and the guard bit; that is, 1.0 will normalize as `m = 1 << 53` and `n = -53`.
```
Comment (fbstj) [inline src/math/generic/fma.rs:235]:
```suggestion
    /// and the guard bit; that is, 1.0 will normalize as `m = 1 << 53` and `e = -53`.
```

**Comment by tgross35** [comment]
🎉 tests pass at 9756810df2dc33f50df67ebfbd9172ee2e9ac747

File: src/math/generic/fma.rs (hunk 1 above)
Comment (tgross35) [reply src/math/generic/fma.rs]: Thanks!

**Comment by tgross35** [comment]
Looks like there is a bug unfortunately. In doing some increased edge cases tests, got the following.

```
thread 'mp_edge_case_fmaf128' panicked at crates/libm-test/tests/multiprecision.rs:17:48:
called `Result::unwrap()` on an `Err` value:
    input:    (0x00007ffffffffffffffffffffffffff7, 0x3ffeffffffffffffffffffffffffffff, 0x00008000000000000000000000000009)
    as hex:   (0x0.7ffffffffffffffffffffffffff7p-16382, 0x1.ffffffffffffffffffffffffffffp-1, 0x0.8000000000000000000000000009p-16382)
    as bits:  (0x00007ffffffffffffffffffffffffff7, 0x3ffeffffffffffffffffffffffffffff, 0x00008000000000000000000000000009)
    expected: 0x00010000000000000000000000000000 0x1.0000000000000000000000000000p-16382 0x00010000000000000000000000000000
    actual:   0x407e0000000000000000000000000000 0x1.0000000000000000000000000000p+127 0x407e0000000000000000000000000000

Caused by:
    TryFromIntError(()): ulp of 85719628837551469319409963899094564864 exceeds u32::MAX
```

**Comment by tgross35** [comment]
Fixed in https://github.com/rust-lang/libm/pull/497
